### PR TITLE
feat(join_no_agg): joining without aggregate

### PR DESCRIPTION
**Pull request by thanh-nguyen-dang**
join without aggregation

### CODE REVIEW

File: tube/etl/indexers/aggregation/translator.py [hunk 1]
```diff
@@ -125,19 +125,39 @@ def get_direct_children(self, root_df):
...
-    def get_joining_props(self, translator, joining_index):
+    def get_joining_props(self, translator, joining_index, with_agg_func):
         """
         Get joining props added by an additional join between indices/documents
         :param joining_index: Joining index created from parser
         :return:
         """
         props = []
         for r in joining_index.getting_fields:
+            if not (with_agg_func ^ (r.fn is None)):
```
Comment (philloooo) [inline tube/etl/indexers/aggregation/translator.py]: why need bit operator here? what's the difference from using just `or`?

Comment (thanh-nguyen-dang) [reply tube/etl/indexers/aggregation/translator.py]: it is equivalent to `(with_agg_func and r.fn is None) or (not with_agg_func and r.fn is not None)`. The way of writing `and` and `or` is more complex

File: tube/etl/indexers/base/parser.py [hunk 2]
```diff
@@ -13,7 +13,17 @@ def __init__(self, mapping, model):
...
-        PropFactory.adding_prop(self.doc_type, '{}_id'.format(self.doc_type), '', [], None, (str,))
+        PropFactory.adding_prop(self.doc_type, '{}_id'.format(self.doc_type), '', [],
+                                src_node=None, src_index=None, fn=None, prop_type=(str,))
+
+    def update_prop_types(self):
+        props = PropFactory.get_prop_by_doc_name(self.doc_type).values()
+        for p in props:
+            if p.type is None:
+                prop = PropFactory.get_prop_by_name(p.src_index, p.src)
+                print (p.src_index)
```
Comment (philloooo) [inline tube/etl/indexers/base/parser.py]: remove?

Comment (thanh-nguyen-dang) [reply tube/etl/indexers/base/parser.py]: sure

File: tube/etl/indexers/aggregation/translator.py (hunk 1 above)
Comment (philloooo) [reply tube/etl/indexers/aggregation/translator.py]: it's so hard to read though, can we just make this return `{props_with_fn: [], props_without_fn: []}` and call it once?

Comment (thanh-nguyen-dang) [reply tube/etl/indexers/aggregation/translator.py]: yeah, good suggestion. I updated

**Pull request merged**